### PR TITLE
[ARROW][GH-34147] Adding testfile for dictionary and crc

### PR DESCRIPTION
**Pull request by mapleFU**
The file is generated by parquet-mr master.

Here I add 3 files for checking checksums, there schema are all:

```avro
message m {
  required int64 long_field;
  required binary binary_field;
}
```

And there contents are checksum. Which is generated by a modified `PageChecksumDataGenerator` with 1000 rows

### CODE REVIEW

**Comment by mapleFU** [comment]
@pitrou @wgtmac Mind take a look?

**Comment by pitrou** [comment]
@wjones127 Would you like to review this?

**Comment by mapleFU** [comment]
(Previously rle is wrong, because parquet-mr `ExampleParquetWriter` use v1 as default, I've fixed it yet

**Comment by mapleFU** [comment]
@wjones127 ping :)

File: data/README.md [hunk 1]
```diff
@@ -41,6 +41,11 @@
 | bloom_filter.bin                               | deprecated bloom filter binary with binary header and murmur3 hashing |
 | bloom_filter.xxhash.bin                        | bloom filter binary with thrift header and xxhash hashing    |
 | nan_in_stats.parquet                           | statistics contains NaN in max, from PyArrow 0.8.0. See note below on "NaN in stats".  |
+| rle-dict-snappy-checksum.parquet                 | compressed INT32 and STRING columns in v2 dictionary pages with a matching CRC |
```
Comment (wgtmac) [inline data/README.md:44]: Does it matter whether the file format is v1 or v2? It seems to me that they are the same dictionary pages except the encoding name. I'd suggest to add less files to keep this repo small.

Comment (mapleFU) [reply data/README.md:44]: Okay, previously, I though we can test format-V1 and format-V2 files. But exactly, v2 is enough for us.

File: data/README.md [hunk 2]
```diff
@@ -41,6 +41,9 @@
...
+| plain-dict-uncompressed-checksum.parquet         | uncompressed INT32 and STRING columns in v1 dictionary pages with a matching CRC |
+| rle-dict-uncompressed-corrupt-checksum.parquet   | uncompressed INT32 and STRING columns in v2 dictionary pages with a mismatching CRC |
```
Comment (wgtmac) [inline data/README.md:46]:
```suggestion
| rle-dict-snappy-checksum.parquet                 | compressed and dictionary-encoded INT32 and STRING columns in format v2 with a matching CRC |
| plain-dict-uncompressed-checksum.parquet         | uncompressed and dictionary-encoded INT32 and STRING columns in format v1 with a matching CRC |
| rle-dict-uncompressed-corrupt-checksum.parquet   | uncompressed and dictionary-encoded INT32 and STRING columns in format v2 with a mismatching CRC |
```

File: data/README.md [hunk 3]
```diff
@@ -111,6 +114,24 @@ The detailed structure for these files is as follows:
   [ Column "b" [ Page 0 [correct crc] | Uncompressed Contents ][ Page 1 [bad crc] | Uncompressed Contents ]]
   ```
 
+The schema for the `*-dict-*-checksum.parquet` test files is:
+* `data/rle-dict-snappy-checksum.parquet`:
+  ```
+  [ Column "long_field" [ Dict Index Page [correct crc] | Compressed PLAIN Contents ][ Page 0 [correct crc] | Compressed RLE_DICTIONARY Contents ]]
+  [ Column "binary_field" [Dict Index Page [correct crc] | Compressed PLAIN Contents ][ Page 0 [correct crc] | Compressed RLE_DICTIONARY Contents ]]
```
Comment (wgtmac) [inline data/README.md:121]:
```suggestion
  [ Column "long_field" [ Dict Page [correct crc] | Compressed PLAIN Contents ][ Page 0 [correct crc] | Compressed RLE_DICTIONARY Contents ]]
  [ Column "binary_field" [ Dict Page [correct crc] | Compressed PLAIN Contents ][ Page 0 [correct crc] | Compressed RLE_DICTIONARY Contents ]]
```

File: data/README.md [hunk 4]
```diff
@@ -111,6 +114,24 @@ The detailed structure for these files is as follows:
...
+  ```
+
+* `data/plain-dict-uncompressed-checksum.parquet`:
+  ```
+  [ Column "long_field" [ Dict Index Page [correct crc] | Uncompressed PLAIN_DICTIONARY(DICT) Contents ][ Page 0 [correct crc] | Unompressed PLAIN_DICTIONARY Contents ]]
+  [ Column "binary_field" [Dict Index Page [correct crc] | Uncompressed PLAIN_DICTIONARY(DICT) Contents ][ Page 0 [correct crc] | Uncompressed PLAIN_DICTIONARY Contents ]]
```
Comment (wgtmac) [inline data/README.md:127]:
```suggestion
  [ Column "long_field" [ Dict Page [correct crc] | Uncompressed PLAIN_DICTIONARY(DICT) Contents ][ Page 0 [correct crc] | Uncompressed PLAIN_DICTIONARY Contents ]]
  [ Column "binary_field" [ Dict Page [correct crc] | Uncompressed PLAIN_DICTIONARY(DICT) Contents ][ Page 0 [correct crc] | Uncompressed PLAIN_DICTIONARY Contents ]]
```

File: data/README.md [hunk 5]
```diff
@@ -111,6 +114,24 @@ The detailed structure for these files is as follows:
...
+  ```
+
+* `data/rle-dict-uncompressed-corrupt-checksum.parquet`:
+  ```
+  [ Column "long_field" [ Dict Index Page [bad crc] | Uncompressed PLAIN Contents ][ Page 0 [correct crc] | Uncompressed RLE_DICTIONARY Contents ]]
+  [ Column "binary_field" [Dict Index Page [bad crc] | Uncompressed PLAIN Contents ][ Page 0 [correct crc] | Uncompressed RLE_DICTIONARY Contents ]]
```
Comment (wgtmac) [inline data/README.md:133]:
```suggestion
  [ Column "long_field" [ Dict Page [bad crc] | Uncompressed PLAIN Contents ][ Page 0 [correct crc] | Uncompressed RLE_DICTIONARY Contents ]]
  [ Column "binary_field" [ Dict Page [bad crc] | Uncompressed PLAIN Contents ][ Page 0 [correct crc] | Uncompressed RLE_DICTIONARY Contents ]]
```

**Review by wgtmac — CHANGES_REQUESTED** [review]

**Review by wgtmac — APPROVED** [review]
LGTM. Thanks!

**Comment by wgtmac** [comment]
@pitrou @wjones127 Do you have any comment?

**Comment by mapleFU** [comment]
By the way, shouldn't we squash and merge, rather than merge directly?

**Pull request merged**